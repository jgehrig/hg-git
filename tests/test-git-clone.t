Load commonly used test logic
  $ . "$TESTDIR/testutil"

  $ git init gitrepo
  Initialized empty Git repository in $TESTTMP/gitrepo/.git/
  $ cd gitrepo
  $ echo alpha > alpha
  $ git add alpha
  $ fn_git_commit -m 'add alpha'
  $ echo beta > beta
  $ git add beta
  $ fn_git_commit -m 'add beta'

  $ cd ..

  $ hg clone gitrepo hgrepo | grep -v '^updating'
  importing git objects into hg
  2 files updated, 0 files merged, 0 files removed, 0 files unresolved
  $ hg -R hgrepo log --graph
  @  changeset:   1:7fe02317c63d
  |  bookmark:    master
  |  tag:         default/master
  |  tag:         tip
  |  user:        test <test@example.org>
  |  date:        Mon Jan 01 00:00:11 2007 +0000
  |  summary:     add beta
  |
  o  changeset:   0:ff7a2f2d8d70
     user:        test <test@example.org>
     date:        Mon Jan 01 00:00:10 2007 +0000
     summary:     add alpha
  

we should have some bookmarks
  $ hg -R hgrepo book
   * master                    1:7fe02317c63d
  $ hg -R hgrepo gverify
  verifying rev 7fe02317c63d against git commit 9497a4ee62e16ee641860d7677cdb2589ea15554

test for ssh vulnerability

  $ hg clone 'git+ssh://-oProxyCommand=rm${IFS}nonexistent/path' | grep -v 'destination\|pulling from'
  abort: potentially unsafe hostname: '-oProxyCommand=rm${IFS}nonexistent'
  [1]
  $ hg clone 'git+ssh://%2DoProxyCommand=rm${IFS}nonexistent/path' | grep -v 'destination\|pulling from'
  abort: potentially unsafe hostname: '-oProxyCommand=rm${IFS}nonexistent'
  [1]
  $ hg clone 'git+ssh://fakehost|shellcommand/path' | grep -v 'destination\|pulling from'
  abort: potentially unsafe hostname: 'fakehost|shellcommand'
  [1]
  $ hg clone 'git+ssh://fakehost%7Cshellcommand/path' | grep -v 'destination\|pulling from'
  abort: potentially unsafe hostname: 'fakehost|shellcommand'
  [1]
