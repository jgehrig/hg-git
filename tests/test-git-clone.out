Initialized empty Git repository in gitrepo/.git/

importing Hg objects into Git
Counting objects: 6, done.
Compressing objects:  33% (1/3)   Compressing objects:  66% (2/3)   Compressing objects: 100% (3/3)   Compressing objects: 100% (3/3), done.
Total 6 (delta 0), reused 0 (delta 0)
importing Git objects into Hg
at: 0/2
updating working directory
2 files updated, 0 files merged, 0 files removed, 0 files unresolved
@  changeset:   1:7bcd915dc873
|  tag:         master
|  tag:         default/master
|  tag:         tip
|  user:        test <test@example.org>
|  date:        Mon Jan 01 00:00:11 2007 +0000
|  summary:     add beta
|
o  changeset:   0:3442585be8a6
   user:        test <test@example.org>
   date:        Mon Jan 01 00:00:10 2007 +0000
   summary:     add alpha

% we should have some bookmarks
 * master                    1:7bcd915dc873
